backend/app/services/interfaces/email_service_provider.py

```diff
@@ -0,0 +1,25 @@
+from abc import ABC, abstractmethod
+
+class IEmailServiceProvider(ABC):
```

Comment:
Thanks for adding this, I think this is a good idea if we decide to swap out Email providers or decide to mock the provider.